prusti-viper/src/encoder/mir/procedures/encoder/mod.rs

```diff
@@ -840,11 +840,41 @@ impl<'p, 'v: 'p, 'tcx: 'v> ProcedureEncoder<'p, 'v, 'tcx> {
             }
             mir::Operand::Copy(source) => {
                 let encoded_source = self.encoder.encode_place_high(self.mir, *source)?;
+
+                let mut lifetime_var: Option<vir_high::VariableDecl> = None;
+                if let vir_high::Expression::Deref(vir_high::Deref {
+                    base:
```

Comment:
`base.get_type()` should give the reference type with the correct `LifetimeConst`.

Reply:
done, thats much nicer